apm-sniffer/apm-sdk-plugin/logger-plugin/src/main/java/org/apache/skywalking/apm/plugin/logger/define/Log4j2LoggerInstrumentation.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.logger.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.NameMatch;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class Log4j2LoggerInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+    private static final String ENHANCE_CLASS = "org.apache.logging.log4j.spi.AbstractLogger";
+    private static final String INTERCEPT_CLASS = "org.apache.skywalking.apm.plugin.logger.Log4j2LoggerInterceptor";
+    private static final String[] INTERCEPT_METHODS = {"trace", "debug", "info", "error", "warn", "fatal"};
+
+    @Override
+    protected ClassMatch enhanceClass() {
+        return NameMatch.byName(ENHANCE_CLASS);
+    }
+
+    @Override
+    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        return new ConstructorInterceptPoint[0];
+    }
+
+    @Override
+    public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
+        return new InstanceMethodsInterceptPoint[]{
+                new InstanceMethodsInterceptPoint() {
+                    @Override
+                    public ElementMatcher<MethodDescription> getMethodsMatcher() {
+                        return named(INTERCEPT_METHODS[0])
+                                .or(named(INTERCEPT_METHODS[1]))
+                                .or(named(INTERCEPT_METHODS[2]))
+                                .or(named(INTERCEPT_METHODS[3]))
+                                .or(named(INTERCEPT_METHODS[4]))
+                                .or(named(INTERCEPT_METHODS[5]));
```

Comment:
From the performance perspective, this is not a good practice. You have to check the method name repeatedly on the runtime. The better way to check this on the bootstrap stage, intercepting the necessary methods.

Reply:
I know there would be a challenge as you could have a complex set of rules. At least, consider https://github.com/apache/skywalking/pull/5914#discussion_r532331110.

Reply:
> From the performance perspective, this is not a good practice. You have to check the method name repeatedly on the runtime. The better way to check this on the bootstrap stage, intercepting the necessary methods

We can determine which insertion points to set according to the method category in the configuration file, right? @wu-sheng @BFergerson . The code just like this:
**LogbackLoggerInstrumentation.class**
```java
@Override
    public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
        if (CONFIG == null) {
            return null;
        }
        return new InstanceMethodsInterceptPoint[]{
                new InstanceMethodsInterceptPoint() {

                    @Override
                    public ElementMatcher<MethodDescription> getMethodsMatcher() {
                        List<String> levelList = CONFIG.getUpeerLevelList(CONFIG.getLevel());
                        ElementMatcher.Junction<MethodDescription> canInstrumentMethods = named(levelList.get(0));
                        for (int i = 1; i < levelList.size(); i++) {
                            canInstrumentMethods = canInstrumentMethods.or(named(levelList.get(i)));
                        }
                        return canInstrumentMethods;
                    }

                    @Override
                    public String getMethodsInterceptor() {
                        return INTERCEPT_CLASS;
                    }

                    @Override
                    public boolean isOverrideArgs() {
                        return false;
                    }
                }
        };
    }
```
**ContextConfig.class**
```java
public List<String> getUpeerLevelList(LogLevel level) {
            List<String> levelList = new ArrayList<>();
            for (LogLevel l : LogLevel.values()) {
                if (level.getPriority() >= l.getPriority()) {
                    levelList.add(l.toString());
                }
            }
            return levelList;
        }
```